frontend-v2/src/features/trial/DoseRow.tsx

```diff
@@ -0,0 +1,194 @@
+import { Delete } from "@mui/icons-material";
+import { IconButton, TableCell, TableRow, Typography } from "@mui/material";
+import { FC, useEffect, useMemo } from "react";
+import FloatField from "../../components/FloatField";
+import IntegerField from "../../components/IntegerField";
+import UnitField from "../../components/UnitField";
+import {
+  DoseRead,
+  Protocol,
+  UnitRead,
+  useDoseRetrieveQuery,
+  useDoseUpdateMutation,
+} from "../../app/backendApi";
+import { Control, useForm } from "react-hook-form";
+import useDirty from "../../hooks/useDirty";
+import useInterval from "../../hooks/useInterval";
+
+type Props = {
+  baseUnit?: UnitRead;
+  control: Control<Protocol, any, Protocol>;
```

Comment:
The amount unit field uses the protocol control, otherwise this form could be completely self-contained, with its own dose controller for each dose row.